### PR TITLE
add pick_signal_channels to baserawio.py

### PR DESCRIPTION
**Pull request by sin-mike**
@samuelgarcia 
following #728 I've added a function to the `baserawio.py`, keeping the options for ordering and strict selecting the channels in other functions

### CODE REVIEW

File: neo/rawio/baserawio.py [hunk 1]
```diff
@@ -421,39 +421,113 @@ def get_group_channel_indexes(self):
         else:
             return [None]
 
-    def channel_name_to_index(self, channel_names):
+    def channel_name_to_index(self, channel_names, user_order=False, strict_selection=True):
```
Comment (samuelgarcia) [inline neo/rawio/baserawio.py:424]: very good

File: neo/rawio/baserawio.py [hunk 2]
```diff
@@ -421,39 +421,113 @@ def get_group_channel_indexes(self):
...
+            channel_indexes = np.array(channel_indexes)
+
+        if strict_selection:
+            assert len(channel_indexes) == len(channel_names), \
+                'selection {} doesn\'t match header names {}'.format(channel_names, chnames)
+
         return channel_indexes
 
-    def channel_id_to_index(self, channel_ids):
+    def channel_id_to_index(self, channel_ids, user_order=False, strict_selection=True):
```
Comment (samuelgarcia) [inline neo/rawio/baserawio.py:450]: very good

**Comment by samuelgarcia** [comment]
Hi @sin-mike.
Sorry for the super super long delay.
I prefer a lot this idea becaise it is generic.
However, I am wondering how this ```pick_signal_channels``` could return another reader and not modify inplace the actual reader.
I think the inplace modification can be dangerous.
What do you think ?

**Comment by sin-mike** [comment]
> Hi @sin-mike.
> Sorry for the super super long delay.
> I prefer a lot this idea becaise it is generic.
> However, I am wondering how this `pick_signal_channels` could return another reader and not modify inplace the actual reader.
> I think the inplace modification can be dangerous.
> What do you think ?

Hi Samuel,

I've been using this feature since then in this exact version, and I didn't observe any drawbacks in our processing pipelines. However, I understand your concern. My suggestion then is to make a protected field `_header`, so that the function can change only the public `header`, while the `_header` is kept untouched. And make a `self.reset_header` function

Mike

**Comment by samuelgarcia** [comment]
Hi @sin-mike 
sorry again for this super long delay.

I like a lot the  `user_order=False, strict_selection=True` but I think it is not possible to include the `pick_signal_channels` (inplace modification) in the neo master/

I think this will lead to some border effect for some rawios even if if works for you.

Maybe you you remove remove this `pick_signal_channels` right now so I could merge the `user_order=False, strict_selection=True` and then you could make another PR that mimic the  `pick_signal_channels(..)` but instead of inplace modification it could just duplicate the the rawio instance and mofify the header. It would be less dangerous.

What do you think ?

**Comment by samuelgarcia** [comment]
Hi @sin-mike.
Sorry for the discontinuous discussion about this old PR.

We will release neo 0.9.0 in september.
Do you think you could a bit more work to test my last idea or would you prefer to prefer to postpone this PR in september.


My last idea is basically to have something like this:

```python
import neo

rawreader = neo.rawio.MyFormetIO(filename=...)  # this have all channel
# this is a copy of the rawreader but with less channels
subrawreader = neo.rawio.tools.pick_signal_channels(rawreader)
```

This would avoid an inplace modification which is dangerous in my mind.

What do ou think ?

**Comment by samuelgarcia** [comment]
Hi Mike,
as the relase date is approaching I propose to postpone this to 0.10.
Cheers

**Comment by samuelgarcia** [comment]
Dear Mike.
I have work a bit rawio API here #949 

The first change you propose here is not possible (in place channel modification).
As I said having another class doing it would be totally ok.

The second change ` strict_selection=True` is no more relevant in the rawio refactoring becaise ids have to be unique inside a stream and this is tested always.

About `user_order=False` I think this is safer to always keep order.

So I close this PR.

Thanks again for you participation in neo.

**Pull request closed without merging**